### PR TITLE
feat: use consolited enum for group status from libsession-util

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: src/groups/meta_group_wrapper.cpp [hunk 1]
```diff
@@ -396,7 +396,11 @@ Napi::Value MetaGroupWrapper::memberGetAllPendingRemovals(const Napi::CallbackIn
     return wrapResult(info, [&] {
         std::vector<session::config::groups::member> allMembersRemoved;
         for (auto& member : *this->meta_group->members) {
-            if (member.is_removed()) {
+            auto memberStatus = member.status();
+            if (memberStatus == session::config::groups::member::Status::removed_unknown ||
+                memberStatus == session::config::groups::member::Status::removed ||
```
Comment (yougotwill) [inline src/groups/meta_group_wrapper.cpp]: I hate that Status is the only thing capitalized 😂 Not your problem but just need to let someone know

File: src/groups/meta_group_wrapper.hpp [hunk 2]
```diff
@@ -25,43 +26,61 @@ struct toJs_impl<member> {
...
+                break;
+
+            // removed statuses
+            case session::config::groups::member::Status::removed_unknown:
+                obj["memberStatus"] = toJs(env, "REMOVED_UNKNOWN");
+                break;
+            case session::config::groups::member::Status::removed:
+                obj["memberStatus"] = toJs(env, "REMOVED_MEMBER");
+                break;
+            case session::config::groups::member::Status::removed_including_messages:
```
Comment (yougotwill) [inline src/groups/meta_group_wrapper.hpp]: We don't need to change this now but these references to variables are pretty damn long. Is there no way to make them shorter via some different importing syntax? Just asking out of interest.

File: types/groups/groupmembers.d.ts [hunk 3]
```diff
@@ -13,32 +13,33 @@ declare module 'libsession_util_nodejs' {
   };
 
   type MemberStateGroupV2 =
-    | 'INVITE_NOT_SENT' // as soon as we've scheduled that guy to be invited, but before we've tried sending the invite message
-    | 'INVITE_FAILED'
-    | 'INVITE_SENT'
+    | 'INVITE_UNKNOWN' // fallback case
```
Comment (yougotwill) [inline types/groups/groupmembers.d.ts]: Maybe use docstrings so we can see the description in session-desktop?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: src/groups/meta_group_wrapper.cpp (hunk 1 above)
Comment (Bilb) [reply src/groups/meta_group_wrapper.cpp]: Status is the name of the class enum, hence capitalized. 
The others are namespaces or values in the enum

File: src/groups/meta_group_wrapper.hpp (hunk 2 above)
Comment (Bilb) [reply src/groups/meta_group_wrapper.hpp]: done

File: types/groups/groupmembers.d.ts (hunk 3 above)
Comment (Bilb) [reply types/groups/groupmembers.d.ts]: Added on docstring on the type itself as docstring do not work on separate strings of a union. This looks ok to me 
image
What do you think?

Comment (yougotwill) [reply types/groups/groupmembers.d.ts]: Yeah fair enough. that is annoying that it is not supported.

**Review by yougotwill — APPROVED** [review]
LGTM 🚢

**Pull request merged**